src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateDataRetentionService.kt

```diff
@@ -37,4 +39,16 @@ class CertificateDataRetentionService(
                 ?: logger.error { "Certificate with sourceType = $sourceType and sourceReference = $sourceReference not found" }
         }
     }
+
+    @Transactional
+    fun removeInitialRetentionPeriodData(sourceType: SourceType) {
+        logger.info { "Finding certificates to remove initial retention period data from" }
+        with(certificateRepository.findPendingRemovalOfInitialRetentionData(sourceType = sourceType)) {
+            forEach {
+                it.removeInitialRetentionPeriodData()
+                certificateRepository.save(it)
+                logger.info { "Removed initial retention period data from certificate with sourceReference ${it.sourceReference}" }
```

Comment:
So a log statement for each update, should we club these to a set of sourceRef and print log at the end - not a blocker

Reply:
I initially logged a summary of how many were removed but it didn't seem very helpful/informative. I think your suggestion makes sense 👍

Reply:
I've been mulling this over. I think either approach is fine for normal every day processing (as I don't think the numbers will be that high), but not if we ever get to the 150,000 applications in one day magical figure! (which I'm assured we won't but there could still be a high number). Perhaps reverting to a simple summary is the best option.

Reply:
Changed back to a summary 🤦‍♂️